src/analysis/convex/segment.lean

```diff
@@ -306,6 +306,29 @@ begin
   rw [←sub_eq_neg_add, ←neg_sub, hxy, ←sub_eq_neg_add, hzx, smul_neg, smul_comm, neg_add_self]
 end
 
+open affine_map
+
+lemma open_segment_subset_union (c : 𝕜) (x y : E) :
+  open_segment 𝕜 x y ⊆ insert (line_map x y c) (open_segment 𝕜 x (line_map x y c) ∪
+    open_segment 𝕜 (line_map x y c) y) :=
+begin
+  simp only [open_segment_eq_image_line_map, ← maps_to'],
+  rintro a ⟨h₀, h₁⟩,
+  rcases lt_trichotomy a c with hac|rfl|hca,
+  { right, left,
+    have hc : 0 < c, from h₀.trans hac,
+    refine ⟨a / c, ⟨div_pos h₀ hc, (div_lt_one hc).2 hac⟩, _⟩,
```

Comment:
```suggestion
    refine ⟨a / c, ⟨by positivity, (div_lt_one hc).2 hac⟩, _⟩,
```
and in other places.

Reply:
What's wrong with short `*_pos` proofs?

Reply:
Not much, but I thought I would showcase `positivity`!